### DISCUSSION TITLE
Is it possible to do a 'delete' or 'clear' mv-action without toast alert?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#883] [Q&A]
Hi

When using multiple calls to mv-action attribute, that include `delete` or `clear`, it toasts a message with ('deleted' + undo button)

But in some cases, delete/clear are part of a global 'normal' mv-action, where  there's no sense to toast 'alert' messages that let the user undo.

So, my question is: Is there a way to ask mavo to not toast in a specific case? Just on specific mv-action's?

Maybe something like mv-action"...." mv-notoast ?

Also related to #842 : 
@DmitrySharabin , in case it's not possible, wouldn't this usecase be an interesting option to add to your plug-in?

**Comment by GalinhaLX** [comment]
I tried to use CSS, like 

.mv-deleted{
display:none;}

but as expected, it removes all toast, even the ones I want to keep (in case I want to undo delete action)

I'm thinking about a mv-action to apply a hidden class, then apply my actions, then remove hidden class again.

Or is there a better way to do this?

**Comment by LeaVerou** [comment]
This is a good idea, I've also needed to hide the toast for specific delete actions. Since `delete()` takes a variable number of arguments, it cannot be an argument of that. Perhaps another function? `delete_silent()` or `sneaky_delete()`?

**Comment by GalinhaLX** [comment]
Hi @LeaVerou 

Thanks for your reply

delete_silent() sounds good for me :)

But may I ask you if you're talking about a [function](https://mavo.io/docs/functions/), or a new [mv-action attribute](https://mavo.io/docs/actions#mv-action)?

Most natural (for me) would be to have it as a new mv-action attribute.

**Reply by LeaVerou** [reply]
A function. You may want to combine non-silent deletes with silent deletes, and a function allows you to do that.

**Comment by karger** [comment]
Can we perhaps generalize to a deleteWithMessage since I often don't like the default one?

**Reply by LeaVerou** [reply]
Interesting. Are you looking to change the message globally or per call?

**Reply by karger** [reply]
Well, per call is surely the most flexible.  In my particular case it would be ok to change globally *for a specific collection* since the problem is that I don't like the default object name inferred by mavo---i have a collection of sign-ups called "singupList" so on delete mavo helpfully announces "signup list deleted" which is going to worry users.  But since I only have one delete action it is just a easy dirt me to change it per call as to change it globally.

**Reply by LeaVerou** [reply]
Sounds like the actual problem is that you need to override the label generated by Mavo from the property name, and the delete message is merely a side effect.

**Reply by DmitrySharabin** [reply]
Yeah, I also noticed it. I was even going to suggest a solution we already have in Mavo. :)

https://mavo.io/docs/ui#localization

**Reply by karger** [reply]
In this case yes, but I'm not sure that the solution to the narrower problem would be any simpler/more convenient than a solution to the more general problem.  I don't think I can use localization to fix what I need; I could use it to change "deleted" to something else but not to change the name of the collection.

**Comment by GalinhaLX** [comment]
Could help till such function will be implemented in Mavo:

Found a 'not so good' way to clear a collection without producing a popup message:

`mv-action="set(collection_name,'')"`       note: there are 2x '   inside, just after comma

not so good because it clears all, but inserts empty value in collection {}
means you need to filter empty values when retrieving collection

**Reply by LeaVerou** [reply]
`set(collection_name, list())` should not have the empty value problem. 😄

**Reply by GalinhaLX** [reply]
hourra, even better :)

Thanks @LeaVerou
